clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -0,0 +1,1021 @@
+                    config.getLong(AdminClientConfig.METADATA_MAX_AGE_CONFIG));
+            metrics = new Metrics(new MetricConfig(), new LinkedList<MetricsReporter>(), time);
+            String metricGrpPrefix = "adminclient";
+            channelBuilder = ClientUtils.createChannelBuilder(config);
+            selector = new Selector(config.getLong(AdminClientConfig.CONNECTIONS_MAX_IDLE_MS_CONFIG),
+                    metrics, time, metricGrpPrefix, channelBuilder);
+            networkClient = new NetworkClient(
+                selector,
+                metadata,
+                clientId,
+                100,
+                config.getLong(AdminClientConfig.RECONNECT_BACKOFF_MS_CONFIG),
+                config.getInt(AdminClientConfig.SEND_BUFFER_CONFIG),
+                config.getInt(AdminClientConfig.RECEIVE_BUFFER_CONFIG),
+                config.getInt(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG),
+                time,
+                true,
+                apiVersions);
+            channelBuilder = null;
+            return new KafkaAdminClient(config, clientId, time, metadata, metrics, networkClient);
+        } catch (Throwable exc) {
+            closeQuietly("Metrics", metrics);
+            closeQuietly("NetworkClient", networkClient);
+            closeQuietly("Selector", selector);
+            closeQuietly("ChannelBuilder", channelBuilder);
+            throw new KafkaException("Failed create new KafkaAdminClient", exc);
+        }
+    }
+
+    static KafkaAdminClient create(AdminClientConfig config, KafkaClient client, Metadata metadata) {
+        Metrics metrics = null;
+        Time time = Time.SYSTEM;
+        String clientId = generateClientId(config);
+
+        try {
+            metrics = new Metrics(new MetricConfig(), new LinkedList<MetricsReporter>(), time);
+            return new KafkaAdminClient(config, clientId, time, metadata, metrics, client);
+        } catch (Throwable exc) {
+            closeQuietly("Metrics", metrics);
+            throw new KafkaException("Failed create new KafkaAdminClient", exc);
+        }
+    }
+
+    private KafkaAdminClient(AdminClientConfig config, String clientId, Time time, Metadata metadata,
+                     Metrics metrics, KafkaClient client) {
+        this.defaultTimeoutMs = config.getInt(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG);
+        this.clientId = clientId;
+        this.time = time;
+        this.metadata = metadata;
+        List<InetSocketAddress> addresses = ClientUtils.parseAndValidateAddresses(
+            config.getList(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG));
+        this.metadata.update(Cluster.bootstrap(addresses), Collections.<String>emptySet(), time.milliseconds());
+        this.metrics = metrics;
+        this.client = client;
+        this.runnable = new AdminClientRunnable();
+        this.thread = new Thread(runnable, "AdminClientThread(" + clientId + ")");
```

Comment:
We should probably use `KafkaThread` here. Naming convention should be consistent with other client threads. For the producer we do:

```java
            String ioThreadName = "kafka-producer-network-thread" + (clientId.length() > 0 ? " | " + clientId : "");
            this.ioThread = new KafkaThread(ioThreadName, this.sender, true);
```